CIP-0072/README.md

```diff
@@ -0,0 +1,413 @@
+                    "signature",
+                    "publicKey"
+                  ]
+                }
+              ]
+            }
+          },
+          "required": [
+            "id",
+            "name",
+            "purpose",
+            "type",
+            "versions",
+            "audits"
+          ]
+        }
+      ]
+    }
+  },
+  "required": [
+    "subject",
+    "projectName",
+    "link",
+    "twitter",
+    "category",
+    "subCategory",
+    "description",
+    "releases",
+    "scripts"
+  ]
+}
+```
+
+## Example
+
+```json
+{
+    "subject": "9SYAJPNN",
+    "projectName": "My Project",
+    "link": "https://myProject.app",
+    "twitter": "https://twitter.com/MyProject",
+    "category": "GAMING",
+    "subCategory": "RPG",
+    "description": {
+                   "short": "A story rich game where choices matter"
+    },	
+
+    "releases": [
+        {
+            "releaseNumber": 1,
+            "releaseName": "V1",
+            "auditId": "z5L90f",
+            "scripts": [
+                {
+                    "id": "PmNd6w",
+                    "version": 1
+                }
+            ]
+        }
+    ],
+    "scripts": [
+        {
+            "id": "PmNd6w",
+	    "name": "Marketplace",
+            "purpose": "SPEND",
+	    "type": "PLUTUS",		
+            "versions": [
+                {
+                    "version": 1,
+		    "plutusVersion": 1, 
+	            "fullScriptHash": "711dcb4e80d7cd0ed384da5375661cb1e074e7feebd73eea236cd68192",                     
+                    "scriptHash": "1dcb4e80d7cd0ed384da5375661cb1e074e7feebd73eea236cd68192",
+                    "contractAddress": "addr1wywukn5q6lxsa5uymffh2esuk8s8fel7a0tna63rdntgrysv0f3ms"
+                }    
+            ],
+          "audits": [
+              {
+                "auditId": "z5L90f",
+                "auditor": "Canonical LLC.",
+                "auditLink": "https://github.com/somlinkToAessment",
+                "auditType": "MANUAL",
+                "signature": "0x1234567890abcdef",
+                "publicKey": "0x1234567890abcdef"
+              }
+          ]
+        }
+    ]
+}
+```
+
+### **Scripts Audits**
+As part of the metadata we will be including a list of audits for each DApp. This will be a list of audits that have been performed on the deployed scripts comprising the DApp. The audits will be performed by a third party auditor and will be signed by the auditor. 
+
+Some of the properties included in the audit are:
+
+- **auditType** property will be included that can take one of the following values: MANUAL, AUTOMATED, or MIXED.
+- **auditId** property will be included that will be a unique identifier for the audit.
+- **auditLink** property will be included that will be a link to the audit report.
+- **signature** property will be included that will be a signature of the audit report.
+- **publicKey** property will be included that will be the public key of the auditor.
+
```

Comment:
```
# Certification

This section describes how certification metadata can be included in CIP-72-compliant metadata for DApp registration. It first describes the requirements for DApp certification, and then sets out the options for its inclusion in CIP-72.

Note: In this section ‘certification’ is used to mean the process of providing various kinds of assurance of DApps through the provision of verifiable, trustworthy metadata linking to relevant evidence of assurance, such as audit reports, test run data etc.

It is expected that evidence of various kinds of assurance of DApps is recorded in an immutable and verifiable way on the Cardano blockchain; these forms of assurance are: testing, including property-based testing (level 1), audit (level 2) and formal verification (level 3). Certification is to be provided by certification providers including: testing services (level 1), auditors (level 2), and verification services (level 3).

## Requirements

Certification metadata should be related to the set of scripts that constitute the deployed version of the DApp that has been certified. 

Certification metadata should contain the following information.:

* the set of deployed scripts to which the metadata pertains
* the certification provider
* the hash of the offchain meta-data by the certification provider
* links to relevant evidence off chain, including, as appropriate,
  * source code repository
  * compiler and version used
  * parameter values for parametric validators 
  * audit report	
  * audit summary (max 100 words)
  * test run data
  * formal proof objects. 
  * certified compilation proof object, etc.

The certification metadata should be linked to DApp registration, as outlined in this CIP.  This requires the registration of any certified DApp prior to, or simultaneously with, the submission of the certification metadata. While audit and testing may take place before a DApp is registered or released, certification is only significant once the DApp is released, and so it can take place once the DApp is registered.

Certification metadata should be signed by the certification provider. It is assumed that certification providers make available their public keys for signature verification.

The metadata should be discoverable by all certification stakeholders, including end-users, DApp developers, and ecosystem components, such as light wallets and DApp stores. Information should be indexable by certification provider, DApp developer, DApp and DApp version.

It should be possible for there to be multiple versions of metadata published by the same certification provider for the same version of a DApp, particularly when a (minor) update of the evidence is necessary. In such a case it should be possible to identify the most recent version of the report for that DApp version. It will also be the case that the same DApp may have certification metadata provided by multiple different certification providers.

It should be possible for wallets to identify to users the certification status of a DApp when they are signing a transaction that is being submitted to a deployed DApp.

## Mechanisms

Evidence of certification is to be posted as explained here. Line references here relate to [CIP-0072? | DApp Registration & Discovery](https://github.com/cardano-foundation/CIPs/pull/355/files#top). 


Evidence is to be published as a separate on-chain item, but as a new `type` of CIP-72 registration.
* Use the same mechanism as for certification (lines 88–142), with a new `type`, `CERTIFICATION` (lines 77–80).
* Assumes that the `subject` will be common with the registration and update of the DApp.
* Assume that it has the same value for `transaction_metadatum_label`, 1667 (lines 143–147).
* This option requires a mechanism to link the certification data to the DApp: this is done via the subject field of the metadata (“the DApp”) and the set of deployed scripts (“the DApp version”).

Advantages
* This item can be managed by the certificate provider independently of other parties. 
* This item can be concerned with certification data only.
* This item is identified by the same label as registration data.
* The metadata will only require update if there is a change in the certification information for the given set of deployed scripts.

Disadvantages
* Requires DApp stores to implement the linking mechanism described above.

Question 
* Will using this typing mechanism require the type `UPDATE` to be split into `REG-UPDATE` and `CERT-UPDATE`?


## General discussion 
1. The model of update outlined in CIP-72, using an explicit `type` for updates, seems to be a particular version of the more general mechanism described in [CIP-26](https://cips.cardano.org/cips/cip26/). It would be clearer to adopt the CIP-26 model for managing metadata, rather than being mechanism agnostic but having to reinvent some aspects of it,
1. From the certification perspective, it makes no sense to audit single contracts. All audit and certification activity is related to DApps, that is to sets of deployed scripts. This needs to be reflected in CIP-72, which links audits to scripts, individually. Of course, it is possible to link from a single script to an audit via the set(s) of scripts to which it belongs (giving the context for the audit).
1. Certificates in CIP-72 are classified as `MANUAL`, `AUTOMATED`, `MIXED`. Preferably this should be levels 1, 2 and 3, fitting with the widely discussed approach to assurance.
1. There needs to be a canonical mechanism for identifying a set of scripts. Two options:
  * Order the script hashes, concatenate them, and then hash the result.
  * Order the scripts according to script names, hash them, concatenate and hash.

```

Reply:
Suggest certification of assurance evidence as outlined above.